src/main/java/com/artipie/asto/Key.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Storage key.
+ *
+ * @since 0.6
+ */
+public interface Key {
+
+    /**
+     * Key.
+     * @return Key string
+     */
+    String string();
+
+    /**
+     * Key from something.
+     * @since 0.6
+     */
+    final class From implements Key {
+
+        /**
+         * Parts.
+         */
+        private final Iterable<String> parts;
+
+        /**
+         * Ctor.
+         * @param parts Parts
+         */
+        public From(final String... parts) {
+            this(Arrays.asList(parts));
```

Comment:
Can you make the constructors code-free?

Reply:
@Sammers21 should be fixed in https://github.com/artipie/asto/pull/20/commits/5adcf2b2ac85bc3b765d35931ad50ec760fe8a03